frontend/reactnative/lib/Auth.js

```diff
@@ -0,0 +1,60 @@
+import Forge from 'node-forge';
+
+import * as Config from './Config.js';
+
+let userKey;
+
+export function wipeUserId(){
+	Config.eraseSetting("userKey").then(function(result) {
+		// log if erase was sucessful or not
+		console.log("Erase sucess: " + result);
+	});
+}
+	
+export async function loadUserId(){
+	// load key
+	Config.retrieveSetting("userKey").then(function(result){
+		// if no key is stored locally
+		if(!result){
+			
+			console.log("no key found, generating a new one");
+			// keep generating keys untill a valid one is found
+			var key;
+			do{
+				key = genUniqueString();
+			}while(!isKeyFree(key));
+			
+			// save valid key to phone
+			Config.storeSetting("userKey", key).then(function(result) {
+				if(!result){
+					console.error("failed to store userKey");
+				}
+			});
+			
+			userKey = key;
+		}else{ 
+			userKey = result;
+		}
+	});
+	
+}
+	
+export function genUniqueString(){
+	//TODO: pad with userAccountInfo
```

Comment:
Create an issue for this TODO if you haven't already done so

Reply:
I have created the Issue #51